### PR TITLE
do perf tests on other branches

### PR DESCRIPTION
**Pull request by TyOverby**

### CODE REVIEW

**Comment by TyOverby** [comment]
@dotnet-bot test ci please

**Comment by TyOverby** [comment]
@mmitche: could you take a look at this?  I feel like I'm doing something wrong...

**Comment by TyOverby** [comment]
cc: @rchande

File: perf.groovy [hunk 1]
```diff
@@ -42,7 +42,7 @@ def generate(boolean isPr) {
     if (isPr) {
         TriggerBuilder prTrigger = TriggerBuilder.triggerOnPullRequest()
         prTrigger.permitOrg('Microsoft')
-        prTrigger.permitOrg('dotnet')
+        prTrigger.permitOrg(branch)
         prTrigger.setCustomTriggerPhrase("(?i)^\\s*(@dotnet-bot\\s+)?(re)?test\\s+perf(\\s+please)?\\s*\$" )
         prTrigger.triggerForBranch('master');
```
Comment (mmitche) [inline perf.groovy]: I think you want the branch here and not on permitOrg

**Comment by TyOverby** [comment]
@dotnet-bot test ci please

File: perf.groovy (hunk 1 above)
Comment (TyOverby) [inline perf.groovy]: let's pretend this never happened

**Comment by TyOverby** [comment]
image

Looks like it's still only generating things for master.

**Comment by mmitche** [comment]
@TyOverby It won't generate them in different folders, but that job will apply to all the branches you specified.

**Comment by mmitche** [comment]
@TyOverby Why not just merge a commit that enables perf changes into the other branches that require perf testing?  That is probably the easiest way to do this.

**Comment by TyOverby** [comment]
@mmitche: not sure what you mean.  The perf.groovy file exists in all the branches that we care about: https://github.com/dotnet/roslyn/tree/dev15.0.x

**Comment by mmitche** [comment]
@TyOverby Ahh, then you just want to add lines for those branches in the repo list: https://github.com/dotnet/dotnet-ci/blob/master/jobs/data/repolist.txt#L70

**Comment by TyOverby** [comment]
@mmitche: could you explain what the "master" string in the perf.groovy script does?  seems like it should be more generic.

**Comment by mmitche** [comment]
@TyOverby It creates a folder for that branch name, a generator script inside that watches that branch fpor changes to perf.groovy, and passes the branch name (and other parameters) to that generator.  This allows for branches to diverge as necessary.

**Comment by TyOverby** [comment]
@mmitche should this change then?

File: perf.groovy [hunk 2]
```diff
@@ -44,7 +44,7 @@ def generate(boolean isPr) {
         prTrigger.permitOrg('Microsoft')
         prTrigger.permitOrg('dotnet')
         prTrigger.setCustomTriggerPhrase("(?i)^\\s*(@dotnet-bot\\s+)?(re)?test\\s+perf(\\s+please)?\\s*\$" )
-        prTrigger.triggerForBranch('master');
```
Comment (mmitche) [inline perf.groovy]: All changed in this file get reverted except for this.  Here you pass branchName instead (variable from https://github.com/dotnet/roslyn/pull/17327/files#diff-bc9a87b2e014aa8266aab4a55fbbc50cL10)

**Comment by TyOverby** [comment]
@dotnet-bot test ci please

**Comment by TyOverby** [comment]
@mmitche: ok, cool!  That was my other hunch!  Should be good now.

**Comment by mmitche** [comment]
Yeah LGTM now.  That change will need to be merged to the other target branches, since they are set up incorrectly at the moment.

**Comment by TyOverby** [comment]
@mmitche 

Cool, I'll cherry-pick this over.  Thanks!

**Comment by TyOverby** [comment]
@mmitche Could I get a signoff?

**Pull request merged**